services/context/context.go

```diff
@@ -267,3 +268,14 @@ func (ctx *Context) JSONError(msg any) {
 		panic(fmt.Sprintf("unsupported type: %T", msg))
 	}
 }
+
+func (ctx *Context) JSONServerError(msg string, err error) {
```

Comment:
Do not do so. Are you sure about what you are doing or it really works?

Reply:
Do you have ideas about handling internal server error if the response should be JSON in backend?

I think it is a bit strange to return bad request status when internal error occurred.
And in frontend, no matter what happened it will always reload the page so it will not show anything.
![image](https://github.com/go-gitea/gitea/assets/18380374/775c31e9-0855-46ea-a44b-8d1a3d6ddff1)

Reply:
The idea is: design some framework mechanisms carefully and rewrite.

Reply:
I have left a comment above:
https://github.com/go-gitea/gitea/pull/30696#discussion_r1586974696

I think we can ignore this issue in this PR, and do it later as your suggested.

Reply:
Or I can revert the changes around the returned 500 and leave it for other PRs since it's not tight with this PR.

Reply:
e9f58eb5f9ed664e3156692467918c6b1ca6093d reverted